java/kotlin-extractor/src/main/kotlin/KotlinFileExtractor.kt

```diff
@@ -1188,22 +1188,16 @@ open class KotlinFileExtractor(
                             id
 
                 val extReceiver = f.extensionReceiverParameter
-                val idxOffset = if (extReceiver != null) 1 else 0
-                val fParameters = overriddenAttributes?.valueParameters ?: f.valueParameters
+                val fParameters = (overriddenAttributes?.valueParameters ?: f.valueParameters) + listOfNotNull(extReceiver)
```

Comment:
Is this correct? The extension receiver is added to the end of the list and below we get the first item in `extendedType = paramTypes[0]`. I think some of the failing tests might also be due to this mixup in the order..

Reply:
Doh, fixed.